From d893aa42ed1afe2a47e105c3562383968f8b83c2 Mon Sep 17 00:00:00 2001
Message-Id: <d893aa42ed1afe2a47e105c3562383968f8b83c2.1441918786.git.yunwei.zhang@intel.com>
In-Reply-To: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
References: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
From: "Razia, RaziaX" <raziax.razia@intel.com>
Date: Tue, 8 Sep 2015 09:31:54 +0530
Subject: [PATCH 09/10] Audio support for HDMI and DP on CHT HR

Audio support in hdmi hot_plug event
Reading audio_reg_base value moved from hdmi_audio_probe
to had_process_hot_plug

Change-Id: I3ae55aeaca4dbf8b633ef13205478726c37ca79a
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-18059
Signed-off-by: Razia, RaziaX <raziax.razia@intel.com>
---
 sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c    |  7 -------
 sound/drivers/intel_hdmi/intel_mid_hdmi_audio_if.c | 16 +++++++++++++++-
 2 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
index e0b1f9d..d8a3caa 100644
--- a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
+++ b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio.c
@@ -1975,13 +1975,6 @@ static int hdmi_audio_probe(struct platform_device *devptr)
 	}
 	intelhaddata->hw_silence = 1;
 
-	/* Query display driver for audio register base */
-	if (intelhaddata->reg_ops.hdmi_audio_get_register_base
-			(&intelhaddata->audio_reg_base)) {
-		pr_err("Unable to get audio reg base from Display driver\n");
-		goto err;
-	}
-
 	intelhaddata->ops = &had_ops_v2;
 	return retval;
 
diff --git a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio_if.c b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio_if.c
index b64f5b8..f84fcb02 100644
--- a/sound/drivers/intel_hdmi/intel_mid_hdmi_audio_if.c
+++ b/sound/drivers/intel_hdmi/intel_mid_hdmi_audio_if.c
@@ -382,7 +382,16 @@ int had_process_hot_plug(struct snd_intelhad *intelhaddata)
 	spin_unlock_irqrestore(&intelhaddata->had_spinlock, flag_irqs);
 
 	pr_debug("Processing HOT_PLUG, buf_id = %d\n", buf_id);
-
+	 /* Query display driver for audio register base */
+	if (intelhaddata->reg_ops.hdmi_audio_get_register_base
+		(&intelhaddata->audio_reg_base)) {
+			pr_err("Unable to get audio reg base from Display driver\n");
+			goto err;
+	}
+	if(intelhaddata->audio_reg_base == 0){
+		pr_err("audio reg base value is NULL\n");
+		goto err;
+	}
 	/* Safety check */
 	if (substream) {
 		pr_debug("There should not be active PB from ALSA\n");
@@ -395,6 +404,10 @@ int had_process_hot_plug(struct snd_intelhad *intelhaddata)
 	had_build_channel_allocation_map(intelhaddata);
 
 	return retval;
+err:
+	pm_runtime_disable(intelhaddata->dev);
+	intelhaddata->dev = NULL;
+	return retval;
 }
 
 int had_process_hot_unplug(struct snd_intelhad *intelhaddata)
@@ -437,6 +450,7 @@ int had_process_hot_unplug(struct snd_intelhad *intelhaddata)
 	spin_unlock_irqrestore(&intelhaddata->had_spinlock, flag_irqs);
 	kfree(intelhaddata->chmap->chmap);
 	intelhaddata->chmap->chmap = NULL;
+	intelhaddata->audio_reg_base = 0;
 	pr_debug("%s: unlocked -> returned\n", __func__);
 
 	return retval;
-- 
1.9.1

